aws/resource_aws_cognito_identity_provider.go

```diff
@@ -25,13 +27,22 @@ func resourceAwsCognitoIdentityProvider() *schema.Resource {
 			"attribute_mapping": {
 				Type:     schema.TypeMap,
 				Optional: true,
+				Computed: true,
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: validation.StringLenBetween(1, 32),
+				},
```

Comment:
In the API reference, this refers to this validation as:

> Key Length Constraints: Minimum length of 1. Maximum length of 32.

Using `ValidateFunc` within `Elem` I believe would be validating the map values, not the keys.

```suggestion
				Elem: &schema.Schema{
					Type: schema.TypeString,
				},
```